src/energyplus/ForwardTranslator/ForwardTranslateScheduleFixedInterval.cpp

```diff
@@ -161,6 +161,11 @@ namespace energyplus {
       unsigned int start = 0;
       if (secondShift == 0) {
         start = 1;
+        // JJR: interval lengths of at least one day shouldn't shift the start of the loop, right? why would we ever start with the second element of the values vector?
+        if (modelObject.intervalLength() == 1440) {  // do this for the one day interval; for interval more than one day, sorry you're on your own
+          start = 0;
+          lastDate -= dayDelta;
+        }
```

Comment:
If we want to support multiple days:

`#include <cmath>  // For std::modf`

```suggestion
        // JJR: interval lengths of at least one day shouldn't shift the start of the loop, right?
        // Why would we ever start with the second element of the values vector?
        const double intervalLength = modelObject.intervalLength();
        double integralPart = 0.0;
        if (std::modf(intervalLength, &integralPart) == 0.0) {
          // The intervalLength is actually an int, not a double
          const int intervalLengthAsInt = static_cast<int>(integralPart);
          // If this is an interval representing one or more days
          if (intervalLengthAsInt % 1440 == 0) {
            start = 0;
            const int nDays = intervalLengthAsInt / 1440;
            lastDate -= dayDelta * nDays;
          }

```

Reply:
According to the new ft test for weekly interval, we'll need more changes/updates than just this.

Reply:
So, I think here the ScheduleFixedInterval.hpp is dumb.

The IDD

```diff
  N1, \field Interval Length
+       \type integer
       \required-field
       \units minutes
       \minimum 1
```


Why is it a double here?

https://github.com/NREL/OpenStudio/blob/9b1c0755f7dad4febeaf69813d2b0f795a55dfef/src/model/ScheduleFixedInterval.hpp#L72

Reply:
I'd vote for an API break here, changing the return type to be the correct one: int. So we can get rid of that std::modf (that I suggested in the first place, I know)

Reply:
What about `setIntervalLength`? Change it to accept only int (instead of double)?

Reply:
Looks like you got it covered. Note to add entry in release notes about the API breaking changes.